mls-rs-crypto-webcrypto/src/ec/ecdh.rs

```diff
@@ -0,0 +1,239 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// Copyright by contributors to this project.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+use mls_rs_core::crypto::{CipherSuite, HpkePublicKey, HpkeSecretKey};
+use mls_rs_crypto_traits::{Curve, DhType};
+
+use der::{oid::ObjectIdentifier, Any};
+use js_sys::Uint8Array;
+use wasm_bindgen_futures::JsFuture;
+use web_sys::{CryptoKey, EcdhKeyDeriveParams, SubtleCrypto};
+
+use super::der_private_key::{generate, DerPrivateKey};
+use crate::{get_crypto, key_type::KeyType, CryptoError};
+
+#[derive(Clone)]
+pub(crate) struct Ecdh {
+    curve_name: &'static str,
+    curve: Curve,
+    named_curve: Any,
+}
+
+impl Ecdh {
+    pub fn new(cipher_suite: CipherSuite) -> Option<Self> {
+        let curve = Curve::from_ciphersuite(cipher_suite, false)?;
+
+        match curve {
+            Curve::P256 => Some(Self {
+                curve_name: "P-256",
+                curve,
+                named_curve: ObjectIdentifier::new_unwrap("1.2.840.10045.3.1.7").into(),
```

Comment:
Might be better to use the oid crate to get these constants?

Reply:
Using the oid crate now. We may consider doing the same for RustCrypto?

Reply:
Yeah for sure. Otherwise it's hard to know if it's correct